rs/bitcoin/kyt/src/fetch.rs

```diff
@@ -225,17 +226,22 @@ pub trait FetchEnv {
                             .into();
                         }
                     }
-                    Pending => continue,
-                    HighLoad | NotEnoughCycles => break,
+                    Pending => {}
+                    HighLoad => {
+                        high_load = true;
+                    }
+                    NotEnoughCycles => {
+                        not_enough_cycles = true;
+                    }
```

Comment:
Maybe I'm missing something here, but shouldn't we break out of the for loop L.206 in those cases to stop processing further `input`?

Reply:
There is actually no need to break out the loop earlier. Breaking out the loop early will only mean the array `futures` is more likely to be empty. So semantically there is a subtle difference, but in practice it shouldn't matter.